pkg/pushsync/pushsync.go

```diff
@@ -231,40 +231,41 @@ func (ps *PushSync) handler(ctx context.Context, p p2p.Peer, stream p2p.Stream)
 
 	// forwarding replication
 	storedChunk := false
-	if ps.warmedUp() && ps.topologyDriver.IsWithinDepth(chunkAddress) {
-
-		verifiedChunk, err := ps.validStamp(chunk, ch.Stamp)
-		if err != nil {
-			ps.metrics.InvalidStampErrors.Inc()
-			ps.logger.Warningf("pushsync: forwarder, invalid stamp for chunk %s", chunkAddress.String())
-		} else {
-			chunk = verifiedChunk
-			_, err = ps.storer.Put(ctx, storage.ModePutSync, chunk)
-			if err != nil {
-				ps.logger.Warningf("pushsync: within depth peer's attempt to store chunk failed: %v", err)
-			} else {
-				storedChunk = true
+	defer func(stored *bool) {
```

Comment:
not sure why it is needed to do all this pointer juggling. if the variable is defined before the closure, you can still modify its value later and just inspect the variable value. no need for pointers